src/repositories/memory/handles.repository.ts

```diff
@@ -62,14 +62,16 @@ class MemoryHandlesRepository implements IHandlesRepository {
                       const handle = HandleStore.get(name);
                       if (handle) {
                           if (search && !handle.name.includes(search)) return agg;
+                          if (handles && !handles.includes(handle.name)) return agg;
                           agg.push(handle);
                       }
                       return agg;
                   }, [])
                 : HandleStore.getHandles().reduce<StoredHandle[]>((agg, handle) => {
-                      if (!search || (search && (handle.name.includes(search) || handle.hex.includes(search)))) {
-                          agg.push(handle);
-                      }
+                      if (search && !(handle.name.includes(search) || handle.hex.includes(search))) return agg;
+                      if (handles && !(handles.includes(handle.name) || handles.includes(handle.hex))) return agg;
+
```

Comment:
I think these two can combine.

Also, I think doing the hex search this way could break for Handles that ARE hex